src/main/java/de/tum/in/www1/artemis/web/rest/AssessmentResource.java

```diff
@@ -35,4 +60,67 @@ void validateExercise(Exercise exercise) throws BadRequestAlertException {
             throw new BadRequestAlertException("The course belonging to this exercise does not exist", getEntityName(), "courseNotFound");
         }
     }
+
+    /**
+     * checks if the user can override an already submitted result. This is only possible if the same tutor overrides before the assessment due date
+     * or if an instructor overrides it.
+     *
+     * If the result does not yet exist or is not yet submitted, this method returns true
+     *
+     * @param submission the submission that might include an existing result which would include information about the assessor
+     * @param exercise the exercise to which the submission and result belong and which potentially includes an assessment due date
+     * @param user the user who initiates a request
+     * @param isAtLeastInstructor whether the given user is an instructor for the given exercise
+     * @return true of the the given user can override a potentially existing result
+     */
+    protected boolean isAllowedToOverrideExistingResult(Submission submission, Exercise exercise, User user, boolean isAtLeastInstructor) {
```

Comment:
Maybe the name of this method and the one below can be slightly changed to differentiate them from the method with the same name in the assessment service that has slightly different preconditions? For the latter, a concrete existing result is passed in as a parameter while the former do not actually require a result to exist to return a useful boolean value. Maybe something like `isAllowedToCreateOrUpdateResult` or `isAllowedToCreateOrOverrideResult`?

Reply:
good point, will adapt it